docs/dev/design/telemetry.rst

```diff
@@ -0,0 +1,269 @@
+         "name": "asn1crypto",
+         "platform": "noarch",
+         "version": "1.4.0"
+      },
+      {
+         "base_url": "https://conda.anaconda.org/conda-forge",
+         "build_number": 3,
+         "build_string": "3",
+         "channel": "conda-forge",
+         "dist_name": "python-3.5.4-3",
+         "name": "python",
+         "platform": "linux-64",
+         "version": "3.5.4"
+      }
+   ]
+
+APT packages
+~~~~~~~~~~~~
+
+We can get the list from the config file,
+or we can list the packages installed with ``dpkg --get-selections``.
+That command would list all pre-installed packages as well, so we may be getting some noise.
+
+.. code-block::
+
+   $ dpkg --get-selections
+
+   adduser                                         install
+   apt                                             install
+   base-files                                      install
+   base-passwd                                     install
+   bash                                            install
+   binutils                                        install
+   binutils-common:amd64                           install
+   binutils-x86-64-linux-gnu                       install
+   bsdutils                                        install
+   build-essential                                 install
+
+Python
+~~~~~~
+
+We can get the Python version from the config file when using a Python environment,
+and from the ``conda list`` output when using a Conda environment.
+
+OS
+~~
+
+We can infer the OS version from the build image used in the config file,
+but since it changes with time, we can get it from the OS itself:
+
+.. code-block::
+
+   $ lsb_release --description
+   Description:    Ubuntu 18.04.5 LTS
+   # or
+   $ cat /etc/issue
+   Ubuntu 18.04.5 LTS \n \l
+
+Format
+~~~~~~
+
+The final information to be saved would consist of:
+
+- organization: the organization id/slug
+- project: the project id/slug
+- version: the version id/slug
+- build: the build id, date, length, status.
+- user_config: Original user config file
+- final_config: Final configuration used (merged with defaults)
+- packages.pip: List of pip packages with name and version
+- packages.conda: List of conda packages with name, channel, and version
+- packages.apt: List of apt packages
+- python: Python version used
+- os: Operating system used
+
+.. code-block:: json
+
+   {
+     "organization": {
+       "id": 1,
+       "slug": "org"
+     },
+     "project": {
+       "id": 2,
+       "slug": "docs"
+     },
+     "version": {
+       "id": 1,
+       "slug": "latest"
+     },
+     "build": {
+       "id": 3,
+       "date/start": "2021-04-20-...",
+       "length": "00:06:34",
+       "status": "normal",
+       "success": true,
+       "commit": "abcd1234"
+     },
+     "config": {
+       "user": {},
+       "final": {}
+     },
+     "packages": {
+        "pip": [{
+           "name": "sphinx",
+           "version": "3.4.5"
+        }],
```

Comment:
I'd like to know if with this "list of dictionaries" we can query 

* projects using sphinx==3.4.5
* projects using sphinx>=2.0.0

I guess we can, but it may be good to double-check it.

Reply:
We could do basic things like "start with `2.`", not sure about something more complex (we could use an > operator, but that will depend on the ascii ordering)